Automattic/wp-calypso
# Reader: move teams to redux using thunks

Reduxified the reader-teams flux store.

Will need to be hooked up to components in a followup PR.

Note: this depends on https://github.com/Automattic/wp-calypso/pull/10856 landing

#### To Test:
- run npm run test-client client/state/selectors and npm run test-client client/state/reader/teams
- make sure calypso build & loads with no errors in the console.
- Also verify that teams is added to the state tree
